fix(query-typeorm): Fix RelationQueryBuilder not properly naming  aliases via DriverUtils.buildAlias

carlkuesters opened the pull request:
This is needed because DriverUtils.buildAlias ensures the alias name is valid by shortening it in case of it exceeding the maximum length supported by the driver (e.g. Postgres only supports alias names up to 63 characters). Also, using this utility is the right thing to do and the class already seems to do it for some parts.

In our project, one specific @ManyToMany relation was not working properly and just returned 0 results, which was traced down to this oversight. Postgres automatically shortened the returned column name, so the lookup that nestjs-query did didn't find a result and just silently returned nothing (Which might also be an improvement area). This behaviour can be very dangerous I think as it might also affect cascading modifications and unknowingly delete data.

See original discussion in the TypeORM PR that added this shortening/hashing feature: https://github.com/typeorm/typeorm/pull/3514

This is my first PR for this project, I assume this is the most active fork of nestjs-query since the original one got "abandoned" (I hope everything is fine for the original author). Please let me know if there is anything I have to do for this PR - I tried using the correct commit message syntax. Thanks for keeping the project alive!

TriPSs commented:
Massive thanks for the PR! 

Could you checking the linting? Can you also validate that the e2e tests still pass? I see there is an issue in the CI that they are not running correctly.

carlkuesters commented:
Hi @TriPSs, I corrected the linting.
I can't see the previous E2E result (as I squashed everything in a single new commit I guess), but as far as I saw it passed and linting was the only thing that failed. I can't start docker-compose at the moment locally, so let's wait what the build says.

TriPSs commented:
Thanks for the PR, sorry for the slow merge wanted to double check if de e2e test where passing correctly as the CI did not run them.
